src/sentry/models/organizationmember.py

```diff
@@ -603,3 +604,15 @@ def is_only_owner(self) -> bool:
             .exists()
         )
         return is_only_owner
+
+
+def organization_member_post_delete(instance: OrganizationMember, **kwargs):
+    region_outbox = None
+    with transaction.atomic():
+        region_outbox = instance.outbox_for_update()
+        region_outbox.save()
+    if region_outbox:
+        region_outbox.drain_shard(max_updates_to_drain=10)
+
+
+post_delete.connect(organization_member_post_delete, sender=OrganizationMember)
```

Comment:
If we do the outbox message creation during a `post_delete` signal it wouldn't be in the same transaction as the record operation. Could we override `delete()` on `OrganizationMember` instead?